.github/workflows/deploy.yml

```diff
@@ -0,0 +1,23 @@
+jobs:
+  docker-image:
+    if: contains( github.ref, 'master')
+    runs-on: ubuntu-latest
+    container: docker:stable
+    
+    services:
+      docker:
+        image: docker:stable-dind
+        env:
+          DOCKER_DRIVER: overlay2
+
+    steps:
+      - name: Check out repositrory code
+        uses: actions/checkout@v2
+
+      - name: Docker commands
+        run: |
+          - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
```

Comment:
Please use the three GitHub Actions from https://docs.docker.com/ci-cd/github-actions/